File: Business Logic Layer(Backend)/Reservation/Reservation.Infra/Service/CreditService.cs
using Reservation.Core.Data;
using Reservation.Core.Service;
using System;
using System.Collections.Generic;
using System.Text;
using Reservation.Core.Repository;

namespace Reservation.Infra.Service
{
    public class CreditService : ICreditService
    {
        private readonly ICreditRepositroy _creditRepository;
        public CreditService(ICreditRepositroy creditRepository)
        {
            _creditRepository = creditRepository;
        }

        public bool DeleteCreditByCreditId(decimal creditId)
        {
            return _creditRepository.DeleteCreditByCreditId(creditId);
        }

        public List<Credit> GetAllCredit()
        {
            return _creditRepository.GetAllCredit();
        }

        public List<Credit> GetCreditByCreditNumber(string creditNumber)
        {
            return _creditRepository.GetCreditByCreditNumber(creditNumber);
        }

        public bool InsertCredit(Credit credit)
        {
            return _creditRepository.InsertCredit(credit);
        }

        public bool UpdateCredit(Credit credit)
        {
            return _creditRepository.UpdateCredit(credit);
        }

        public bool UpdeteAmountByCreditNumber(Credit credit)
        {
            return _creditRepository.UpdeteAmountByCreditNumber(credit);
        }
    }
}

File: Presentation Layer (frontend)/Reservation_angular/src/app/service/credit.service.ts

import { CartItem } from "src/data/cartItem";
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { Login } from 'src/data/login';
import { environment } from 'src/environments/environment';
import { SharedModule } from '../shared/shared.module';
import jwt_decode from 'jwt-decode';
import { Register } from 'src/data/register';
import { Credit } from "src/data/credit";
@Injectable({
  providedIn: 'root'
})
export class CreditService {

  headerDict={
    'Content-Type':'application/json',
    'Accept':'application/json'
    }
 requestOptions={
      headers:new HttpHeaders(this.headerDict)
      }
 constructor(public http:HttpClient,private route:Router) { }

    InsertCredit(credit :Credit){

       return this.http.post(environment.apiUrl+"Credit/InsertCredit/",credit,this.requestOptions)
    }
    UpdateCredit(credit :Credit){

       return this.http.put(environment.apiUrl+"Credit/UpdateCredit/",credit,this.requestOptions)
    }
    DeleteCreditByCreditId(creditId:number){

       return this.http.delete(environment.apiUrl+"Credit/DeleteCreditByCreditId?creditId="+creditId,this.requestOptions)
    }
    GetAllCredit(){

       return this.http.get(environment.apiUrl+"Credit/GetAllCredit",this.requestOptions)

    }
    UpdeteAmountByCreditNumber(credit:Credit){

        return this.http.put(environment.apiUrl+"Credit/UpdateCredit/",credit,this.requestOptions)

    }
    GetCreditByCreditNumber(creditNumber:string){

        return this.http.get<any[]>(environment.apiUrl+"Credit/GetCreditByCreditNumber?creditNumber="+creditNumber,this.requestOptions)

    }
        }